js_july_23/Users/Sandeep/Homework/loops HW 14082023.html

```diff
@@ -0,0 +1,135 @@
+        //     First value a_0 = a
+        //     Second value a_1 = a + d
+        //     Third value a_2 = a + 2*d
+
+        //     nth value a_n-1 = a + (n-1)*d
+
+        //     E.g.,
+        //     for initial value 3, common difference 5, number of values as 4, 
+        //     Print: 3, 8, 13, 18
+
+        const a = 3;
+        // Initialization of Value
+        const d = 5;
+        // Common difference
+        const n = 4;
+        // Number of values
+
+        document.write("Arithmetic Progression (AP):<br>");
+
+        for (let i = 0; i < n; i++) {
+            let currentTerm = a + i * d;
+            document.write(currentTerm + ", " + "<br>");
+        }
+
+    </script>
+
+
+    <script>
+        //     2. Print numbers in Geometric progress (GP)
+        // initial value (a), common-ratio (r), and number of values (n)
+
+        // nth value a_n-1 = a * (r raised to the power (n-1)) = a * (r * r * .. n-1 times)
+
+        // E.g., initial value = 3, common ratio = 2, number of values = 4
+        // Print: 3, 6, 12, 24
+
+
+        const a1 = 3;
+        // Initialization of value
+        const r = 2;
+        // Common ratio
+        const n1 = 4;
+        // Number of values
+
+        document.write("Geometric Progression (GP):<br>");
+
+        let currentTerm = a;
+        for (let i = 0; i < n1; i++) {
+            document.write(currentTerm + ", " + "<br>");
+            currentTerm *= r;
+        }
+
+    </script>
+
+    <script>
+
+        // 3. Print all the digits of a number in reverse order.
+        //     E.g., for number 4356
+        //     Print: 6, 5, 3, 4
+
+        const number = 4356;  // The number
+
+        const numberStr = number.toString(); // Convert the number to a string
+        const length = numberStr.length;     // Get the length of the string
+
+        document.write("Digits in reverse order:<br>");
+
+        for (let i = length - 1; i >= 0; i--) {
+            const digit = parseInt(numberStr[i]); // Convert the character back to an integer
+            document.write(digit + ", " + "<br>");
+        }
+
+    </script>
+
+
+    <script>
+        // 4. From years 1 to 2024, print if the year is leap year or not.
+        //     E.g., 
+        //     1 is not a leap year
+        //     2 is not a leap year
+        //     3 is not a leap year
+        //     4 is a leap year
+        //     5 is not a leap year
+        //     ....
+        //     2024 is a leap year
+
+        for (let year = 1; year <= 2024; year++) {
+            if ((year % 4 === 0 && year % 100 !== 0) || (year % 400 === 0)) {
+                document.write(year + " is a leap year<br>");
+            } else {
+                document.write(year + " is not a leap year<br>");
+            }
+        }
+
+    </script>
+    
+    <script>
+
+        const number1 = 16;
+        const sqrt1 = Math.sqrt(number1);
```

Comment:
Though it is correct, Do not use inbuilt library.

Use loops/conditions. 

Sqrt doesn't have to be accurate. Let's say upto 0 decimal digits i.e. only whole numbers.